app/src/test/java/pm/gnosis/heimdall/ui/safe/create/ConfirmRecoveryPhraseViewModelTest.kt

```diff
@@ -0,0 +1,69 @@
+package pm.gnosis.heimdall.ui.safe.create
+
+import io.reactivex.observers.TestObserver
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentCaptor
+import org.mockito.junit.MockitoJUnitRunner
+import pm.gnosis.heimdall.ui.recoveryphrase.ConfirmRecoveryPhraseViewModel
+import pm.gnosis.svalinn.common.utils.DataResult
+import pm.gnosis.svalinn.common.utils.Result
+import pm.gnosis.tests.utils.ImmediateSchedulersRule
+import pm.gnosis.utils.words
+
+@RunWith(MockitoJUnitRunner::class)
+class ConfirmRecoveryPhraseViewModelTest {
+    @JvmField
+    @Rule
+    val rule = ImmediateSchedulersRule()
+
+    private lateinit var viewModel: ConfirmRecoveryPhraseViewModel
+
+    @Before
+    fun setup() {
+        viewModel = ConfirmRecoveryPhraseTestViewModel()
+    }
+
+    @Test
+    fun testGetMnemonic() {
+        viewModel.setup(RECOVERY_PHRASE)
+        assertEquals(RECOVERY_PHRASE, viewModel.getRecoveryPhrase())
+    }
+
+    @Test
+    fun getIncorrectPositionsWrongRecoveryPhrase() {
+        val testObserver = TestObserver.create<Result<Set<Int>>>()
+        val words = "0 media athlete 3 rocket plate minute obey head toward coach senior".words()
+
+        viewModel.setup(RECOVERY_PHRASE)
+        viewModel.getIncorrectPositions(words).subscribe(testObserver)
+        testObserver.assertValue { it is DataResult && it.data.containsAll(listOf(0, 3)) }
+    }
+
+    @Test
+    fun getIncorrectPositionsCorrectRecoveryPhrase() {
```

Comment:
What about the generation method? Even if it is random we can check that the returned number of words is correct and that the words are part of the recovery phrase